spec/components/blacklight/document_component_spec.rb

```diff
@@ -156,4 +156,40 @@ def call
     component.partial { 'Partials' }
     expect(rendered).to have_content 'Partials'
   end
+
+  describe '.with_collection' do
+    let(:docs) do
+      [
+        SolrDocument.new(id: '123', title_tsim: 'Book1'),
+        SolrDocument.new(id: '456', title_tsim: 'Book2')
+      ]
+    end
+    let(:response) do
+      instance_double(Blacklight::Solr::Response,
+                      documents: docs,
+                      prev_page: nil,
+                      next_page: 2,
+                      total_pages: 3,
+                      aggregations: [])
+    end
+
+    let(:config) do
+      Blacklight::Configuration.new do |config|
+        config.add_index_field 'title_tsim', label: 'Title:'
+      end
+    end
+
+    let(:render) do
+      described_class.with_collection(docs).render_in(view_context)
```

Comment:
We never use this class in this way in the code. Why do we want a test for this invocation?

Reply:
Verifying comes at little cost (this is just demonstrating that), and speaking for myself: If I understand that Blacklight is moving to ViewComponents, I am going to expect that I can learn how to use ViewComponents and apply that knowledge to my Blacklight app. Whether the current, say, index view uses `with_collection` or not seems like less of an issue than whether it might.